### DISCUSSION TITLE
Is langium appropriate to add browser/ide support to an existing language?

### DISCUSSION THREAD

**Discussion by RickeyWard** [eclipse-langium/langium#1736] [Q&A] [answered]
My team used Jexl (Apache Java expression language ) as scripting in a project. There's no tooling available for IDEs to help with jexl and ideally our Monaco web inputs would support it too)

I was wondering if that's an appropriate use case for langium or if it's more tailored specifically for creating new languages.

Thanks!

**Comment by msujew** [comment, marked as answer]
Hey @RickeyWard,

I wasn't aware of Jexl yet, but it looks like a good use case for Langium. In fact, existing languages are often times easier to support using Langium compared to novel languages, because they usually have a well-defined grammar that can be poured into our grammar format.

Looking closer at Jexl, it seems to import Java types - this isn't a problem for Langium in general (our linking feature is very generic), but getting this to work often times requires some sort of JDK bridge/adapter. We built something like that for [this project](https://github.com/BBx-Kitchen/bbj-language-server/) (see `java-interop` folder for more info), in case you're interested to see what I'm talking about.

The current version of the yeoman generator we offer for langium (see [here](https://langium.org/docs/learn/workflow/install/) and [here](https://langium.org/docs/learn/workflow/scaffold/)) supports creating a VS Code extension and a monaco web-worker for online IDE support.

**Reply by RickeyWard** [reply]
Thank you so much for answer!

**Reply by spoenemann** [reply]
See this conference presentation for an overview of the example project with java-interop mentioned by Mark, and specifically the Java-related challenges:
https://www.youtube.com/watch?v=FD1p7dPHiIw&list=PLmmNK7CRoSWt5oY4OnvZOkrVBQklvxrG8&index=3

**Reply by diegopego** [reply]
Thanks for the answer! I have a similar case. I use Harbour (xbase++ / ca-clipper) and I have a large codebase. Having tooling such as the provided bu langium would be very nice!
